corehq/apps/domain/forms.py

```diff
@@ -954,12 +956,9 @@ def save(self, domain):
 pwd_pattern = re.compile( r"([-\w]){"  + str(min_pwd) + ',' + str(max_pwd) + '}' )
 
 def clean_password(txt):
-    if len(txt) < min_pwd:
-        raise forms.ValidationError('Password is too short; must be at least %s characters' % min_pwd )
-    if len(txt) > max_pwd:
-        raise forms.ValidationError('Password is too long; must be less than %s characters' % max_pwd )
-    if not pwd_pattern.match(txt):
-        raise forms.ValidationError('Password may only contain letters, numbers, hyphens, and underscores')
+    strength = zxcvbn(txt, user_inputs=['commcare', 'hq', 'dimagi', 'commcarehq'])
+    if strength['score'] < 2:
+        raise forms.ValidationError(_('Password is not strong enough. Try making your password more complex.'))
```

Comment:
Instead of just using this blanket message have we considered using the feedback from zxcvbn `result.feedback.warning` and `result.feedback.suggestions`? This could give more specific guidance for the user. If there is no feedback we could default to this message.